### PR TITLE
Plotly

### PR DESCRIPTION
**Pull request by GrantJamesPowell**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -101,30 +101,47 @@ const getTemperatures = () => {
...
   for (let i = 0; i < steps; i++) {
     for (let j = 0; j < steps; j++) {
-      if (temperatures[i * steps + j] > 1000) {temperatures[i * steps + j] = 1000}
-      //const weight = (temperatures[i * steps + j] - 300) / (3000 - 300)
-      //ctx.fillStyle = pickHex("rgb(0, 0, 255)", "rgb(255, 0, 0)", weight);
-      //ctx.fillStyle = `rgb(${temperatures[i * steps + j] % 255}, 100, 100)`
-      ctx.fillStyle = `rgb(${(temperatures[i * steps + j] - 300) / (1000 - 300) * 255}, 0, 0)`
-      ctx.fillRect(i, j, 1, 1)
+      z[i][j] = rosenthal(x[i], y[j], 0, values.initialTemp, values.power, values.thermalConductivity, values.velocity, alpha);
+      if (z[i][j] > 3000) {z[i][j] = 3000;}
```
Comment (GrantJamesPowell) [inline app.js]:
```
Math.clamp(x, lower, upper)
```

File: index.html [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
+    <script src="https://cdn.plot.ly/plotly-latest.min.js"></script>
+
     <link rel="stylesheet" type="text/css" href="style.css">
     <title>Rosenthal Calculator</title>
   </head>
   <body>
     <div class="app">
       <div class="title">Rosenthal Calculator</div>
+
+      <div id='plotlyDiv'></div>
```
Comment (GrantJamesPowell) [inline index.html:12]: Is this where this goes?

File: index.html [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
+
       <div class="graphs">
```
Comment (GrantJamesPowell) [inline index.html:14]: Do we need the other graphs anymore

File: app.js [hunk 4]
```diff
@@ -101,30 +101,47 @@ const getTemperatures = () => {
...
+
+  // get x and y values
+  for (var i = 0; i < size; i++) {
+  x[i] = -values.x + (2 * values.x / size) * i;
+  y[i] = -values.y + (2 * values.y / size) * i;
+  z[i] = new Array(size)
+  }
+
+  // get thermal diffusivity
+  const alpha = values.thermalConductivity / (8000 * 460);
```
Comment (GrantJamesPowell) [inline app.js:121]: I'd pull these into top level constants. Magic numbers are tough

File: app.js [hunk 5]
```diff
@@ -101,30 +101,47 @@ const getTemperatures = () => {
...
-//}
-
 const drawPlots = () => {
-  const temperatures = getTemperatures();
-  const canvas = document.getElementById("xyGraph");
-  const ctx = canvas.getContext("2d");
+
+  const values = formValues();
 
   const steps = 500;
```
Comment (GrantJamesPowell) [inline app.js:108]: I'd pull this into a top level constant

File: app.js (hunk 4 above)
Comment (eheikkenen) [reply app.js:121]: Ya, that was just temporary

File: index.html (hunk 2 above)
Comment (eheikkenen) [reply index.html:12]: I'm not sure

File: index.html (hunk 3 above)
Comment (eheikkenen) [reply index.html:14]: I thought I removed them

**Pull request merged**